server/monitor-types/tailscale-ping.js

```diff
@@ -0,0 +1,25 @@
+const { MonitorType } = require("./monitor-type");
+const { UP } = require("../../src/util");
+
+class TailscalePing extends MonitorType {
+
+    name = "tailscale-ping";
+
+    async check(monitor, heartbeat) {
```

Comment:
I am actually changing the structure, so I built the boilerplate for you. 

Basically, you have to work on this `check()` function, see comments for more details and see #3308 for reference.

Reply:
I would also like to ask, I already have this [existing repository](https://github.com/SGprooo/Uptime-Kuma-Tailscale-Docker) where tailscale works with uptime kuma through an ICMP Ping, my original version of this dockerfile was by fetching the latest tarball binary from Tailscale directly, but it later involved into combining tailscale's image.

I'm wondering in this case, should I implement that (tarball install) into uptime kuma when the user selects a the tailscale monitoring type?

Reply:
I think you should assume that it is for non-Docker setup only and no need to include executables in the docker image, because adding Tailscale support into the main image is actually another challenging task.